core/tests/test_network.py

```diff
@@ -0,0 +1,27 @@
+from testcontainers.core.container import DockerContainer
+from testcontainers.core.docker_client import DockerClient
+from testcontainers.core.network import Network
+
+
+def test_network_gets_created_and_cleaned_up():
+    with Network() as network:
+        docker = DockerClient()
+        networks_list = docker.client.networks.list(network.name)
+        assert networks_list[0].name == network.name
+        assert networks_list[0].id == network.id
+    assert not docker.client.networks.list(network.name)
+
+
+def test_containers_can_communicate_over_network():
+    with Network() as network:
+        with DockerContainer("nginx:alpine-slim").with_name(
```

Comment:
Instead of using `with_name` (which can lead to naming conflicts when parallelizing tests or accidentally reusing names), can we add an API  `with_network_aliases(string[])`, similar to [tc-java](https://java.testcontainers.org/features/networking/#advanced-networking), that uses Docker's `--network-alias` mechanism?